### ISSUE TITLE
[issue] Editing Elementor on iPad

### ISSUE THREAD

**Issue by bodhi1970** [pojome/elementor#3365]
I have the issue that I can not edit a page with Elementor on iPad (Air 2, latest OS) neither with Safari nor with Firefox. Elementor opens and it seems I could use the widgets but I only see the top of the page and can not scroll down to other sections below.

![4ede8d35-8eab-4541-9a1c-dea9f6b9eaaa](https://user-images.githubusercontent.com/5292941/35677548-5671a1d2-0750-11e8-8ea5-b7272c53fa81.jpeg)

**Comment by dasigna** [comment]
i dont think pagebuilders in general were ever made to work on tablets.
is this what you are expecting - seriously?
dont know how that would ever work sufficiently... eighter way. too limited resolution, no hover functinality at all etc. etc.

maybe i am not expecting that much, but from my side i never ever tried to develop any website on a tablet... really dont know how to get this done on such a small limited screen (and only one!) :-)

**Comment by bodhi1970** [comment]
I am not speaking about developing a website on tablet, but though it would be great to edit text or change a picture quickly from on the road when the laptop is not around.

**Comment by bodhi1970** [comment]
... and it is not far away to being able to do that. It’s only the visibility of the full page in the preview window and the ability to scroll. Other things seem to work. I can click on the top section, I can drag and drop a widget to the section and I can edit the section.

**Comment by bainternet** [comment]
@bodhi1970 

It all comes down to the fact that Ipad/tablets and mobile devices are missing key features like `click` and  `hover` so we never really addressed it and I don't see us focusing on this anytime soon.

but yea that would be nice. :)

**Comment by TikiKitchen** [comment]
I would like the ability to use Elementor on a tablet as there are times that I need to quickly design a page before a meeting, and Elementor is a great tool for creating quick click prototyping in a design process, and often I would have to do that on a tablet.

I would never use the iPad or table for development, and don’t consider Elementor a development tool. The development is in how I add functionality through proper coding, extending features, hooking in to key filters, and creating custom short codes that fill out the client specific needs within the codebase. Elementor is just a nice means of providing an additional service to my clients to allow them create their own pages.

**Comment by kkeennaa** [comment]
I have the same issue. I basically dont own a laptop, only an iPad Pro, and sometimes I have to edit a page on the go and cant. Sometimes I’m traveling and I need to fix something for a client. Can’t. It would be great to be able to do it even if its like on a simplified version. 

So much about responsivess.

**Comment by nancilillamb** [comment]
Me too. Since Elementor is a drag&drop builder for non programmers (real programmer write strings of code...), making it work on the device most non programmers/visual driven people use (otherwise, we would’t need the “mobile first” mindset) would be a great idea to make it work on iPad. May bring  a great advantage over other builders.

The only missing thing is the scrolling! Having the “preview” part in a container and make it scroll in it, might do the trick. I’m not sure what click/hover missing’s got to do with that... it’s just not scrolling.

**Comment by micbus** [comment]
Me too. I have a blog and use elementor for my blog post design. It works all great, but I can‘t write posts on business travels or vacations, where I have no laptop with me. Hope you will focus sometimes on this feature. I would even pay for this extra.

**Comment by kkeennaa** [comment]
I think the solution is making a single article page for this case with Pro, but yeah, editing on iOS would be a huge plus. I just had an emergency yesterday and the iPad was the only thing I had in hand.













Julia Braga
Brand Designer & Creative Strategist
The Branding Girl
m: +358403522411
w: thebrandingirl.com  e: julia@thebrandingirl.com





Services & Contact

**Comment by rammyleeds** [comment]
I know people have been saying that Elementor was designed for PC and laptop use - and I get that.  But I totally agree with Bodhi1970 who started off this post merely saying she would like to edit her website on a tablet.  And comments like designa's are completely unhelpful when he/she comments that 'Elementor was never designed for ipads' and 'why would you want to create/edit a website on an ipad anyway'.

The fact is, everything is moving this way.  People want to become mobile.  And be able to do things anywhere.  AND THIS IS TOTALLY ACHIEVABLE.  There are buckets of things you can do on the ipad that you would never of dreamed of doing without a PC.   You can even do payments for instance without your credit card and just with your phone!!!!  So listen up, the world is changing.   A lot of people are moving over to mobile editing.

My point?  I accept Elementor does not claim to work with ipads or other tablets.   But that they should be striving towards this because this is the way things are going.   And besides, most of the functionality of Elementor seems to be available on the ipad - but you simply can't use it because the SCROLL function doesnt work which means you cannot scroll to the bit of the webpage you want to edit!!!  If they fixed the SCROLLING element, perhaps the rest would simply just work!   

And it was the scrolling function that Bodhi1970 was asking about - and whether it was fixable.  So, come on Elementor - strive towards doing better.   You can edit websites with other web tools like Divi and Wix, so Elementor and Beaver Builder - you need to do the same.  Come on, you know you can make it happen.

**Comment by drake-dev** [comment]
I 100% agree with rammyleeds. We’re in 2018 and touch screens have been around for a LONG time. It would be very helpful to add productivity across platforms. It honestly makes no sense to me that there isn’t more functionality on mobile. PC’s and Laptops are going to transition to tablet formats. That’s just the way it is. It would be very nice to have this feature.

**Comment by ntercity** [comment]
Please add workaround elementor editor on mobile. Do not ignore touchscreen user.

**Comment by Hallsworthb** [comment]
I’ll just add a “Me too”. I have a “days to next meeting” on a club website and that needs to be reset. Maybe I need to move it to the top left, in the header, but I shouldn’t need to do that.

**Comment by DanGitOnline** [comment]
With an ipad and bluetooth keyboard, I can copy and paste an element, but can’t delete an element... only thing I’m missing.  Seems possible to do everything else I can think of.  The keyboard has a delete key and backspace key that work in the notes app, Elementor probably needs at least one more keycode to catch to make this work.

**Comment by DigitalLeaves** [comment]
> i dont think pagebuilders in general were ever made to work on tablets.
> is this what you are expecting - seriously?
Actually that what quite a dumb and not very useful remark. Yes, iPads and tablets are becoming the de-facto new computers of our time. And this is only going to grow in the following years. Didn't you know we are in the post-PC era? Check your website's traffic stats. Probably websites will be edited on mobile phones instead of tablets soon.

So yes, I would say that would be a very useful feature. Makes Elementor unusable as of today.

**Comment by dasigna** [comment]
@DigitalLeaves ...
actually theres no respect within your answer, nor did you read my post well enough - obviously.
theres a good chance you are right regarding the upcoming new computer or the post-pc-era. no doubt about that. but that wasnt what i wrote at all.
yet i wrote about the present, and _up till now_ pagebuilders dont seem to be buildt for mobile devices. i'll stick to that.
some day they will be for shure. but thats not the present. so the future yet is still to come for this.
and only as a little sidemark as you proudly claimed youre working on websites exclusively on an ipad pro: congrats. if that works for you its allright. but opposits to that there will be a lot of water flowing down the rivers of the world again until you wont see any pc or workstation on the desks of even web developers.

so you might regard this a very needed 'fix' for elementor, but its actually not buildt with that in mind.
looking on it closely you wont disagree - and seriously thats no point who of us is right or not. its just as it is for now.
i merely have the elementor guys chime in for some clearance ...
maybe i'm wanting a postive answer as much as you would, but once again i dont think theres one _yet_.

for closing this just one little thing as a personal new years wish if you want it to:
please think about your respect and your words twice again in future before posting harsh comments and claim others to be dumb!
possibly i am - but its not up to you to judge, as my comment was not. maybe you wont read such, but this alone wont make it (or even myself) dumb or negative.

cheers.

**Comment by DigitalLeaves** [comment]
Hi @dasigna, I'm glad to know you feel offender and that there was no "respect" in my words, that's exactly how @bodhi1970 should have felt when reading your answer:

> i dont think pagebuilders in general were ever made to work on tablets.
is this what you are expecting - seriously?

Happy new year and relax man! You'll live longer ;)

**Comment by dasigna** [comment]
@DigitalLeaves
allright my lad, youre right. but dont ever get too much arrogant in your kingdom :)

**Comment by DanGitOnline** [comment]
I've tried to make Ipad work and with the right tweaks, I had a system down for copy/delete and ignoring the lack of mouse support, but didn't realize how severe the scrolling bug was in Safari or all the browser shells that seem to be just Safari in a wrapper.  I gave up trying to use the Ipad pro for work stuff and am gettin' by good with the chromebook.

**Comment by gkuehn** [comment]
Seriously? Ì´ve read through all of the feed. With that attitude we would never have left the caves. Why? Because men aren’t thought to live outside caves as my grandgrandgrandgrandgrandgrand..grandmom used to say. As a webdesigner frequently travelling and meeting people outside my office I would love to see elementor working on ipad. Because it is more convenient than to make notes or to carry heavy weight equipment from the office to the people. And in my branch, from media, web & design, photo to illustration, literally everybody has an ipad and loves it because it is a robust and serious instrument. Even the AV media guys are using it to field edit their 4K footage. So the people creating the face of the web changed in larger parts to ipad next to their desktops. For someone with a such a great tool adressing this folk it would be smart not to teach but to consider changing if your peergroup changes. Unless you wan‘t to do the dino thingy.

**Comment by DigitalLeaves** [comment]
> Seriously? Ì´ve read through all of the feed. With that attitude we would never have left the caves. Why? Because men aren’t thought to live outside caves as my grandgrandgrandgrandgrandgrand..grandmom used to say. As a webdesigner frequently travelling and meeting people outside my office I would love to see elementor working on ipad. Because it is more convenient than to make notes or to carry heavy weight equipment from the office to the people. And in my branch, from media, web & design, photo to illustration, literally everybody has an ipad and loves it because it is a robust and serious instrument. Even the AV media guys are using it to field edit their 4K footage. So the people creating the face of the web changed in larger parts to ipad next to their desktops. For someone with a such a great tool adressing this folk it would be smart not to teach but to consider changing if your peergroup changes. Unless you wan‘t to do the dino thingy.

👏🏻👏🏻👏🏻

**Comment by Hallsworthb** [comment]
11months on and no progress. Does Gutenberg allow iPad edits?

**Comment by DigitalLeaves** [comment]
> 11months on and no progress. Does Gutenberg allow iPad edits?
@Hallsworthb It kind of does...ish. But the interface is ugly and not comfortable at all to work on.

**Comment by mehran92** [comment]
can you not add a toggle that disables all drag/click events and adds overflow-y: auto; to the right div? I think this is a great selling point, everything loads fine. I think its a waste to not add a simple work around, specially with the new iPad pro.

**Comment by richtong** [comment]
+1 for this. This (and photo editing) and the only thing holding me back from all touch. All iPad Pro world :-)

**Comment by mtingle777** [comment]
+1 for iPad editing with elementor. It all works except I can’t scroll down to edit sections below the top half of the page.

iPads are the future! Super handy for on the fly editing and minor tweaking.

**Comment by fugazidave** [comment]
Now that iPad OS is on the horizon for this fall, and now that it will finally allow the use of a mouse, can we please, please, please get iPad editing with elementor? There are a ton of us now that use the iPad Pro as our primary computer day to day, and that number is only going to keep growing. I like that I can leave the MBP at home most days, except on the days when I need to work on my sites. It’s really annoying. There’s clearly demand for this, and that demand is only going to increase as time goes by.

**Comment by rammyleeds** [comment]
Yes, everyone is banging on for this request.  Come on Elementor – you are big enough to make this happen.

**Comment by MacDschie** [comment]
Same here. Sometimes, I work with the iPad only, not carrying the laptop. I surely don’t buy a product which is not finished in a way that I can‘t use it on the devices I have to.

**Comment by mehran92** [comment]
Hi guys, i just managed to get scrolling working with the new OS 13 Beta 3 update. 

1. **Install OS 13 Beta** : go here to download, https://beta.apple.com/sp/betaprogram/ here is a video instruction on how to install the new update https://www.youtube.com/watch?v=0AomDpTfZnw

2. Setup wireless mouse, I am using a Logitech MX Master 2S (no dongle needed). Here is a video also on how to connect your mouse https://www.youtube.com/watch?v=OTEDQEhglVA

Once that is done, you can scroll and use a mouse. Even though the scroll wheel works on other apps and website, it won't work for some reason on elementor edit mode yet. So, you will have to hover over the scrollbar and click and drag like you would on your computer (move the mouse all the way to the right side of the screen). I hope this helps, it wont be as smooth as working on a PC, but at least you can work on it now :)

**Comment by rammyleeds** [comment]
Thanks for this.   It didn’t work well for me.   If your using a page builder like Elementor, you still can scroll down.  At a push you can use the new navigator to do things – but it is painful.   The solution is better than what was there before, but didn’t work for me.  How did it work for others?

 

Thanks for the post


Ram, Leeds

**Comment by mehran92** [comment]
@ram

did you try requesting the desktop version? that improved it a little, but you are right. it wont be smooth as a computer. I feel right now it is not at a stage where you can build fast enough on it to use it to work, but it is good enough to edit things on the go. it should get better with ios updates, the scroll wheel didn't work either :/ but what was your main problem? maybe i can help

**Comment by jvkrimpen** [comment]
This is sooo much wanted. But not only on iPad, also (at least) text editing on a mobile is very much needed. If you want to make quick changes while you're on the road this is essential. I understand that real page builder functions on mobile are not very helpful, but at least text editing and for instance selecting/changing a featured image would be minimal.

By the way, there are other page builders that allow you to change, amongst others, the text in the traditional WP editor.

**Comment by kevingraziani** [comment]
+ for iPad editing on elementor

**Comment by josevillo** [comment]
Hi!

I came in looking for a solution to the scroll issue. I wanted to change some images of a gallery, so I read all the posts.

BUT, after that, went back to Elementor and I realized that THE SCROLL WORKS PERFECTLY IN THE IPAD PRO!! AND I AM USING IOS 12.4

SOLUTION: GO TO THE WINDOW YOU WANT TO SCROLL AND TOUCH THE WINDOW VERY CLOSE TO THE BORDER AND MOVE UP OR DOWN. YOU WILL SEE THE LITTLE “LIFTER” APPEARS FOR A WHILE.

YOU HAVE TO TOUCH IN THE WHITE SPACE JUST BETWEEN THE IMAGES OF AND THE BORDER OF THE WINDOW. I DID IT WITH THE APPLE PENCIL BUT IT WORKS WITH THE FINGER TOO TOUGH ITS A LITTLE MORE DIFFICULT. AND IT WORKS ON BOTH SIDES OF THE WINDOW!

HOPE THIS CAN HELP YOU AND SORRY FOR MY BAD ENGLISH.

![5669D773-F309-45E0-9ABC-4A810B76B784](https://user-images.githubusercontent.com/56724180/67091289-a5ba4000-f182-11e9-992d-a30fcdc99c0f.jpeg)

**Comment by rammyleeds** [comment]
Yes, it works a treat.  Thank you.  Although editing is still quite fiddly unless you just want to edit text and do simple things.  For more complicated things like moving elements around – I strongly advise you use your PC.

**Comment by fugazidave** [comment]
So it’s getting closer to being actually usable and I’m doing more and more work on sites from my iPad. However, I’m running into one really annoying thing. Using the iPad Pro and Magic Touchpad, I cannot figure out any way to get the menu that you would normally get when right-clicking or some equivalent (like two-finger tap on my MBP). I would mostly be able to deal with this, but it means I cannot delete any widgets. I can add, but I can’t remove.

Anyone have a workaround for this?
